src/lib/utils/track.ts

```diff
@@ -201,3 +219,59 @@ export interface Track<P = any, S = null, T extends Schema.Global = Schema.Entit
  *      ```
  */
 export const track: Track = _track
+
+/**
+ * A typed page view decorator for a class, for when you _don't_ need to make tracked calls internally
+ *
+ * As an object:
+ *
+ * @example
+ *
+ *      ```ts
+ *      import { pageViewTrack, Schema } from "lib/utils/track"
+ *
+ *       @pageViewTrack({
+ *        context_screen: Schema.PageNames.ConsignmentsWelcome,
+ *        context_screen_owner_slug: null,
+ *        context_screen_owner_type: Schema.OwnerEntityTypes.Consignment,
+ *       })
+ *       export default class Welcome extends React.Component<Props, null> {
+ *
+ *        goTapped = () => this.props.navigator.push({ component: Overview })
+ *
+ *        // It's not optimal to dismiss the modal, but otherwise we get into all sorts of tricky states
+ *        privacyPolicyTapped = () =>
+ *          SwitchBoard.dismissModalViewController(this) &&
+ *          SwitchBoard.presentNavigationViewController(this, Router.PrivacyPage)
+ *         [...]
```

Comment:
Is anything but the class definition and decorator really needed? I have a hard time parsing out the important parts of the example due to the amount of code.

Reply:
yeah, just the definition, I can remove the extras